tests/AllowHalfOpenTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace React\Tests\Socket;
+
+use React\EventLoop\Factory;
+use React\EventLoop\LoopInterface;
+use React\Socket\ConnectionInterface;
+use React\Socket\Connector;
+use React\Socket\TcpServer;
+
+function _log ($message) {
+    echo $message . "\n";
+};
```

Comment:
Why not just use `echo` directly?

Reply:
You are right, it's because _log function was more complex before. Anyway, I plan to remove it when the API is discussed and the PR is ready. Please let me know what do you think of approach in general #126?